roles/openshift_registry/tasks/main.yml

```diff
@@ -12,3 +10,56 @@
     --credentials={{ openshift_master_config_dir }}/openshift-registry.kubeconfig {{ _oreg_images }}
   register: _oreg_results
   changed_when: "'service exists' not in _oreg_results.stdout"
+
+- name: Create temp directory for volume definitions
+  command: mktemp -d /tmp/openshift-ansible-XXXXXXX
+  register: mktemp
+  changed_when: False
+  when: attach_registry_volume | bool
+
+- name: Land registry PersistentVolume definition
+  template:
+    dest: "{{ mktemp.stdout }}/registry-volume.json"
+    src: registry-volume.json.j2
+  when: attach_registry_volume | bool
+
+- name: Land registry PersistentVolumeClaim definition
+  template:
+    dest: "{{ mktemp.stdout }}/registry-claim.json"
+    src: registry-claim.json.j2
+  when: attach_registry_volume | bool
+
+- name: Create registry PersistentVolume
+  command: >
+    {{ openshift.common.client_binary }} create -f {{ mktemp.stdout }}/registry-volume.json
+  register: pv_create_output
+  failed_when: ('already exists' not in pv_create_output.stderr if pv_create_output.stderr else False) or ('created' not in pv_create_output.stdout if pv_create_output.stdout else False)
+  when: attach_registry_volume | bool
+
+- name: Create registry PersistentVolumeClaim
+  command: >
+    {{ openshift.common.client_binary }} create -f {{ mktemp.stdout }}/registry-claim.json
+  register: pvc_create_output
+  failed_when: ('already exists' not in pvc_create_output.stderr if pvc_create_output.stderr else False) or ('created' not in pvc_create_output.stdout if pvc_create_output.stdout else False)
+  when: attach_registry_volume | bool
```

Comment:
There was a email thread recently about this topic and I believe the consensus there was to use direct volume mounts for infra related items, since the complexity of the pv/pvc wasn't needed there and to rely on the pv/pvc mechanism for user deployed applications.

I'm not directly sold on either way, but as we start adding more infra items that need to be backed by persistent storage, it might be nice to be able to tell the user how to dump the objects to get the volumes in use rather than having to try and map through the pv/pvc indirection.